### PR TITLE
Add kitura basic & kitura api

### PR DESCRIPTION
**Pull request by DunnCoding**
This PR aims to add two new Kitura CLI commands. 

The first is `kitura basic` which creates a skeleton Kitura project with an example "Hello World" route. 
The second is `kitura api` which scaffolds a project similar to `kitura init` but includes KituraOpenAPI as well. 

This PR brings the CLI more in line with the Kitura App.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,9 +8,12 @@
     "kitura-create": "kitura-create.js",
     "kitura-idt": "kitura-idt.js",
     "kitura-init": "kitura-init.js",
+    "kitura-basic": "kitura-basic.js",
+    "kitura-api": "kitura-api.js",
     "kitura-kit": "kitura-kit.js",
     "kitura-run": "kitura-run.js",
-    "kitura-sdk": "kitura-sdk.js"
+    "kitura-sdk": "kitura-sdk.js",
+    "helper": "helper.js"
```
Comment (ianpartridge) [inline package.json]: I don't think this should be in `bin` should it?

File: kitura.js [hunk 2]
```diff
@@ -6,11 +6,13 @@ const program = require('commander');
 program
     .version(pjson.version)
     .description('Kitura command-line interface')
-    .command('build', 'build the project in a local container')
+    .command('init', 'scaffold a bare-bones Kitura project')
```
Comment (ianpartridge) [inline kitura.js]: I'm not sure that "bare-bones" is the best way to describe this now.

File: kitura-basic.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env node
+
+const path = require('path');
+const tools = require('./helper');
+
+const gitURL = 'https://github.com/IBM-Swift/generator-swiftserver-projects'
```
Comment (ianpartridge) [inline kitura-basic.js]: I thought we needed to end lines with semi-colons to preserve compatibility with older Node.js versions... I might be wrong though.

Comment (DunnCoding) [reply kitura-basic.js]: Those lines were copied from what we already had. 
`kitura init` didn't end those lines with a semi-colon either however I think we should as it is regarded as good practice.

File: kitura.js (hunk 2 above)
Comment (DunnCoding) [reply kitura.js]: No, bare-bones is not the best description. 

What would you prefer? Cloud ready?

File: kitura-init.js [hunk 4]
```diff
@@ -29,158 +26,27 @@ if (projName.charAt(0) === '.') {
...
-}
-
-
-
 function printHelp() {
     console.log("");
     console.log("  Usage: kitura init [options]");
     console.log("");
-    console.log("  Scaffold a bare-bones Kitura project.");
+    console.log("  Scaffold a basic cloud ready Kitura project.");
```
Comment (ianpartridge) [inline kitura-init.js]: Delete "basic" because we also have `kitura basic` and we don't want to cause any (more) confusion.

File: kitura.js [hunk 5]
```diff
@@ -6,11 +6,13 @@ const program = require('commander');
 program
     .version(pjson.version)
     .description('Kitura command-line interface')
-    .command('build', 'build the project in a local container')
+    .command('init', 'scaffold a basic cloud ready Kitura project.')
+    .command('api', 'scaffold a Kitura project with OpenAPI')
+    .command('basic', 'create a skeleton Kitura project')
```
Comment (ianpartridge) [inline kitura.js]: create -> scaffold

File: kitura-api.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+let args = process.argv.slice(2);
+
+if (args.length > 0) {
+    if (args.indexOf('--help') > -1) {
+        printHelp();
+        process.exit(0);
+    }
+}
+
+let currentDirPath = path.resolve("./");
```
Comment (ianpartridge) [inline kitura-api.js]: This section of the file is common between `init`, `basic` and `api`, can we have more common code so we don't duplicate it?

File: kitura-basic.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env node
+
```
Comment (ianpartridge) [inline kitura-basic.js:2]: Please add the IBM copyright header to every file in this project. It needs to go after the shebang.

**Review by ianpartridge — APPROVED** [review]

**Comment by DunnCoding** [comment]
Closed as being reworked under this [epic](https://github.com/IBM-Swift/kitura-cli/issues/53)

**Pull request closed without merging**